modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/management_bar/components/filters/AutocompleteFilter.js

```diff
@@ -162,162 +156,130 @@ function AutocompleteFilter({
 			return true;
 		}
 
-		const prevValues = prevValue.map((element) => element.value).sort();
-		const newValues = newValue.map((element) => element.value).sort();
+		let changed = false;
 
-		return prevValues.some((element, i) => element !== newValues[i]);
+		const prevValues = prevValue.map((el) => el.value).sort();
+		const newValues = newValue.map((el) => el.value).sort();
+
+		prevValues.forEach((element, i) => {
+			if (element !== newValues[i]) {
+				changed = true;
+			}
+		});
```

Comment:
The old version was better because `some()` will short-circuit as soon as it finds a mismatch, whereas this one will go through the whole list even after finding a mismatch.

Also, I think we probably need a test for length, don't we? How do we know that `prevValues` and `newValues` have the same length? Because if they don't, you can wind up comparing against `undefined` (if `newValues` has fewer items than `prevValues`), or falsely claiming that nothing changed just because `prevValues` is shorter than `newValues` and happens to match the first items.

ie.

```js

let changed = false;

if (prevValue.length === newValue.length) {
   // sort + check etc
}
else {
   changed = true;
}

return changed;
```

Reply:
> The old version was better because some()
Good catch. I think I ended up losing the prior optimisation when applying new stuff from commerce. 

Git diff doesn't show, but there's a check for length already in there https://github.com/liferay-frontend/liferay-portal/blob/c376c4499bd2a66e500eb0685e6e0eff2ec4af33/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/management_bar/components/filters/AutocompleteFilter.js#L155